packages/manager/apps/web/client/app/hosting/offer/upgrade/hosting-offer-upgrade.controller.js

```diff
@@ -110,40 +110,55 @@ angular.module('App').controller(
       );
     }
 
-    getDetachActionsOptions() {
-      return {
-        addons: [
-          {
+    getDetachRequestOptions(serviceId, planCode) {
+      return this.ovhManagerProductOffersActionService
+        .getDetachPlancodeInformationOptions(serviceId, planCode)
+        .then((data) => {
+          const addons = [];
+          if (data.length === 1) {
+            const addonService = data[0].serviceId;
+            const addonPlanCode = data[0].plans[0].planCode;
+
+            addons.push({
+              duration: DETACH_DEFAULT_OPTIONS.durationCode,
+              planCode: addonPlanCode,
+              pricingMode: DETACH_DEFAULT_OPTIONS.pricingMode,
+              quantity: DETACH_DEFAULT_OPTIONS.quantity,
+              serviceId: addonService,
+            });
+          }
+          return {
+            addons,
             duration: DETACH_DEFAULT_OPTIONS.durationCode,
-            planCode: this.model.offer.value,
             pricingMode: DETACH_DEFAULT_OPTIONS.pricingMode,
             quantity: DETACH_DEFAULT_OPTIONS.quantity,
-            serviceId: this.serviceId,
-          },
-        ],
-        duration: DETACH_DEFAULT_OPTIONS.durationCode,
-        pricingMode: DETACH_DEFAULT_OPTIONS.pricingMode,
-        quantity: DETACH_DEFAULT_OPTIONS.quantity,
-      };
+          };
+        });
     }
 
-    getPrices() {
-      if (this.isDetachable) {
-        return this.ovhManagerProductOffersActionService
-          .simulate(
-            this.model.offer.value,
-            this.serviceId,
+    getDetachPrices(serviceId, planCode) {
+      return this.getDetachRequestOptions(serviceId, planCode)
+        .then((detachOptions) => {
+          return this.ovhManagerProductOffersActionService.simulate(
+            planCode,
+            serviceId,
             DETACH_DEFAULT_OPTIONS.type,
-            this.getDetachActionsOptions(),
-          )
-          .then((data) => {
-            const durationsTab = [];
-            const details = angular.copy(data.order);
-            details.duration = DETACH_DEFAULT_OPTIONS.durationText;
-            durationsTab.push(details);
+            detachOptions,
+          );
+        })
+        .then((data) => {
+          const detachPrices = [];
```

Comment:
```suggestion
        return [{
        ...data.order,
        duration: DETACH_DEFAULT_OPTIONS.durationText
        }]
```
no need to create an array variable then push, directly return the value

Reply:
thanks for the review, it is corrected